runtime/src/main/java/org/corfudb/runtime/exceptions/TrimmedException.java

```diff
@@ -1,10 +1,22 @@
 package org.corfudb.runtime.exceptions;
 
+import lombok.Getter;
+import lombok.Setter;
+import org.corfudb.runtime.view.Address;
+
 /**
  * This exception is thrown when a client tries to read an address
  * that has been trimmed.
  */
 public class TrimmedException extends LogUnitException {
+    @Getter
+    @Setter
+    long address = Address.NON_ADDRESS;
+
+    @Getter
+    @Setter
+    long snapshot = Address.NON_ADDRESS;
```

Comment:
This isn't quite what we discussed... I guess it would be okay to store this address, which BTW, is not a snapshot, but the minimum object "version"), but it should provide a boolean (in this case, perhaps just via a method, called isNonrecoverable(), for example), so the client doesn't haven't do compute whether or not it is recoverable.

Frther